### PR TITLE
random sql changes around the codebase

### PR DESCRIPTION
**Pull request by danehrlich1**
https://github.com/openemr/openemr/issues/2035

### CODE REVIEW

**Comment by danehrlich1** [comment]
@bradymiller 

Does the below look ok? It was a slightly unusual statement so wanted to check. Old and New are below. Thanks.

Original Line: https://github.com/openemr/openemr/blob/029ad61cb9de91e0bf3f6958a3c3ef9cd723e14f/library/clinical_rules.php#L1919

New Line:
```
  // check for items
    if (empty($column)) {
        // simple search for any table entries
        $sql = sqlStatementCdrEngine("SELECT * " .
        "FROM ? ".
        "?".
        "WHERE ? = ? ?", array((add_escape_custom($table)), $whereTables, $patient_id_label, $patient_id, $customSQL));
```

File: patients/add_edit_event_user.php [hunk 1]
```diff
@@ -439,64 +439,35 @@
...
-            "1," .(int)$_POST['facility']. ")"); // FF stuff
+                sqlInsert("INSERT INTO openemr_postcalendar_events ( " .
+                "pc_catid, pc_aid, pc_pid, pc_title, pc_time, pc_hometext, " .
+                "pc_informant, pc_eventDate, pc_endDate, pc_duration, pc_recurrtype, " .
+                "pc_recurrspec, pc_startTime, pc_endTime, pc_alldayevent, " .
+                "pc_apptstatus, pc_prefcatid, pc_location, pc_eventstatus, pc_sharing, pc_facility                                                                                  
+                 ) VALUES (                                                                                                                                                         
+                 ?, ?, ?, ?, NOW(), ?, ?, ?, ?                                                                                                                                      
+                 ?, ?, ?, ?, ?, ?, ?, ?, ?, 1, 1, ?)",
+                 array($_POST['form_category'], $provider, $_POST['form_pid'], $_POST['form_title'],
```
Comment (bradymiller) [inline patients/add_edit_event_user.php]: shouldn't above $provider be $_POST['form_provider'] ?

File: patients/add_edit_event_user.php [hunk 2]
```diff
@@ -510,11 +481,10 @@
 // Applied by Cassian Lup (cassian.lup@clinicdr.com)
 
     if (0) {
-        $tmprow = sqlQuery("SELECT count(*) AS count FROM form_encounter WHERE " .
-        "pid = '" . $_POST['form_pid'] . "' AND date = '$event_date 00:00:00'");
+        $tmprow = sqlQuery("SELECT count(*) AS count FROM form_encounter WHERE 
+        pid = ? AND date = ? 00:00:00", array($_POST['form_pid'], $event_date));
```
Comment (bradymiller) [inline patients/add_edit_event_user.php]: This one get everybody at first :)

You need to keep the entire variable:
```
pid = ? AND date = ?", array($_POST['form_pid'], $event_date." 00:00:00"));
```

**Comment by bradymiller** [comment]
hi,
Now we are getting into advanced sql injection :)
You can only bind individual variables and can not bind other stuff (like table names or column names). That being said, we do have a better and more secure mechanism than simply running them through add_escape_custom that uses whitelisting to ensure the actual table or column actually exists. It would look like this:
```
        $sql = sqlStatementCdrEngine("SELECT * " .
        "FROM `" . escape_table_name($table)  . "` " .
        " ". $whereTables. " ".
        "WHERE " . escape_sql_column_name($patient_id_label, array($table))  . "=? " . $customSQL, array($patient_id));
```
Note $whereTables is a string created above and is secure.

escape_table_name() and escape_sql_column_name() are really cool functions and you can see how they work in library/formdata.inc.php

**Pull request closed without merging**